pkg/artifacts/signable.go

```diff
@@ -149,40 +152,56 @@ type image struct {
 }
 
 func (oa *OCIArtifact) ExtractObjects(ctx context.Context, obj objects.TektonObject) []interface{} {
-	log := logging.FromContext(ctx)
 	objs := []interface{}{}
-
-	// TODO: Not applicable to PipelineRuns, should look into a better way to separate this out
-	if tr, ok := obj.GetObject().(*v1beta1.TaskRun); ok {
-		imageResourceNames := map[string]*image{}
-		if tr.Status.TaskSpec != nil && tr.Status.TaskSpec.Resources != nil {
-			for _, output := range tr.Status.TaskSpec.Resources.Outputs {
-				if output.Type == backport.PipelineResourceTypeImage {
-					imageResourceNames[output.Name] = &image{}
-				}
+	if trV1, ok := obj.GetObject().(*v1.TaskRun); ok {
+		var resources v1beta1.TaskResources //nolint:staticcheck
+		shouldReplace := false
+		if serializedResources, ok := trV1.Annotations["tekton.dev/v1beta1-status-taskrunstatusfields-taskspec-resources"]; ok {
```

Comment:
It's not immediately obvious to me what this annotation is / why we need this (I'm assuming this has something to do with Pipeline Resource compat). Can you add some comments explaining what's happening here + links to any context? 🙏

Reply:
TLDR; this code was left as an example and in case I misunderstood what functionality could be deprecated or how to convert this field properly.  In my investigation I believe I now understand this is deprecated field, chains has type hinting which works for v1beta1 and v1 and there is actually no way to properly get this field from v1beta1 -> v1 (-> v1beta1) as the `.Status` values aren't resolved/populated by the Tekton Pipeline conversion webhook.

This is confusing, sorry for not commenting directly in the codebase.  The logic here in actuality should be removed but was kept at this stage as an example to show what would be needed for 1:1 backward compatibility.  The annotation `tekton.dev/v1beta1-status-taskrunstatusfields-taskspec-resources` isn't an actual annotation from Tekton Pipelines but a custom hack/workaround for serializing/deserializing this field from v1beta1 -> v1 -> v1beta to get tests passing and understand what was needed to keep this working.  In practice though this is not possible (v1beta1 -> v1 -> v1beta1) as we only listen for v1 objects and thusly can only do v1 -> v1beta1.  While the Tekton Pipeline Conversion webhook technically will do v1beta1 -> v1 for us the field there isn't populated by the webhook and this info is lost.   For this field chains needs the `.Status` vs `.Spec` field IIUC and the Tekton Pipeline Conversion webhook only converts `.Spec` fields which are not fully resolved.

Reply:
I see now that @JeromeJu is considering updating the Tekton Pipeline conversion logic to support `.Status` fields.  @JeromeJu can you comment on if the `.Status.TaskRunStatusFields.TaskSpec.Resources` field will be supported there?

If so we can just use `ConvertFrom`, keep the functionality and tests and not have any  bespoke logic or comments necessary

Reply:
Yeah, I believe with https://github.com/tektoncd/pipeline/pull/7507, `v1beta1.TaskRun.TaskRunStatusFields.TaskSpec.Resources` should be supported.

Reply:
One question remaining is do we want to export all annotation keys for deprecated fields (for pipeline) in a separate PR?

Reply:
+1 I think it should be fine to not export the annotation keys.  Currently (with the updates made in your tektoncd/pipeline PR - https://github.com/tektoncd/pipeline/pull/7507) the code here no longer needs to manually use such keys.

Reply:
This additionally logic is removed now related to functionality added in https://github.com/tektoncd/pipeline/pull/7507.  Resolving the thread